src/server/auth/server/api_server.go

```diff
@@ -262,7 +355,7 @@ func (a *AuthServer) expiredEnterpriseCheck(ctx context.Context, username string
 
 func (a *AuthServer) userHasExpiredClusterAccessCheck(ctx context.Context, username string) error {
 	// Root User, PPS Master, and any Pipeline keep cluster access
-	if username == auth.RootUser || strings.HasPrefix(username, auth.PipelinePrefix) {
+	if username == auth.RootUser || strings.HasPrefix(username, auth.PipelinePrefix) || strings.HasPrefix(username, auth.InternalPrefix) {
```

Comment:
How good do you feel about a prefix check?  Can my pachyderm username ever be `pipeline:foobar` (not with oauth, but maybe ldap if that's a connector we support)?

Reply:
the only ways customer can mint tokens is either via authenticating in which case we tack on a "user:" prefix to the provided username, or via robots in which case we tack on a "robot:" prefix. We definitely rely on this prefix prepending